back/engines/commercial/insights/app/controllers/insights/web_api/v1/classification_tasks_controller.rb

```diff
@@ -35,17 +35,14 @@ def create
           head :accepted
         end
 
-        def destroy_all
-          tasks = ZeroshotClassificationTasksFinder.new(view.categories).execute.destroy_all
-          status = tasks.map(&:destroyed?).all? ? :ok : :internal_server_error
-          head status
-        end
+        def destroy_tasks
+          tasks = ZeroshotClassificationTasksFinder.new(
+            categories || view.categories, # use all the categories if the query parameter is not provided
+            inputs: inputs
+          ).execute
 
-        def destroy
-          # We find the task via the finder to make sure it's associated with the right view.
-          # [TODO] Actually, nothing prevents a task from being associated to categories from several views.
-          task = ZeroshotClassificationTasksFinder.new(view.categories).execute.find(params[:id])
-          status = task.destroy.destroyed? ? :ok : :internal_server_error
+          Insights::CategorySuggestionsService.new.cancel(tasks)
```

Comment:
Shouldn't we check the response code here too (see comment below) and send the request status according to it?